CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -288,34 +319,36 @@ game:GetService("RunService"):BindToRenderStep("Cursor3D", Enum.RenderPriority.L
 	local ignoreList = { player.Character }
 	local part, endpoint = workspace:FindPartOnRayWithIgnoreList(ray, ignoreList)
 
-	cursor.Visible = false
-	if part ~= hopperbinPart then
+	if part ~= panel.part then
+		for i, v in pairs(Tools) do
+			if v.hovered then
+				v.OnLeave()
+			end
+		end
 		return
 	end
-	cursor.Visible = true
 	
 	local localEndpoint = part:GetRenderCFrame():pointToObjectSpace(endpoint)
-	local x = ((localEndpoint.X / part.Size.X) * 1) + 0.5
-	local y = ((localEndpoint.Y / part.Size.Y) * 1) + 0.5
-	x = 1 - x
-	y = 1 - y
-	cursor.Position = UDim2.new(x, -cursor.AbsoluteSize.x * 0.5, y, -cursor.AbsoluteSize.y * 0.5)
+	local x = 1 - ((localEndpoint.X / part.Size.X) + 0.5)
+	local y = 1 - ((localEndpoint.Y / part.Size.Y) + 0.5)
 	
 	--REMOVE THIS WHEN GUI MOUSELEAVE/MOUSEENTER ARE FIXED
-	local px = cursor.AbsolutePosition.X + cursor.AbsoluteSize.X * 0.5
-	local py = cursor.AbsolutePosition.Y + cursor.AbsoluteSize.Y * 0.5
+	local px = x * panel.gui.AbsoluteSize.X
+	local py = y * panel.gui.AbsoluteSize.Y - 36 --WHY OH WHY
```

Comment:
Alright, I think we can fix this in the engine really quick.
The `WHY OH WHY` is too powerful